packages/cli/src/cms/api/api-client.ts

```diff
@@ -13,54 +14,115 @@ import type {
   UpsertRemoteResponse,
 } from './types';
 
+interface BaseApiClient {
+  request(url: string, options: FetchWithTimeoutOptions): Promise<Response>;
+}
+type CommandOption = 'push' | 'push-status';
+export type SunsetWarning = { sunsetDate: Date; isSunsetExpired: boolean; warning: string };
+export type SunsetWarningsBuffer = SunsetWarning[];
+
 export class ReuniteApiError extends Error {
   constructor(message: string, public status: number) {
     super(message);
   }
 }
 
-class ReuniteBaseApiClient {
-  constructor(protected version: string, protected command: string) {}
-
-  protected async getParsedResponse<T>(response: Response): Promise<T> {
-    const responseBody = await response.json();
+class ReuniteApiClient implements BaseApiClient {
+  public sunsetWarnings: SunsetWarningsBuffer = [];
 
-    if (response.ok) {
-      return responseBody as T;
-    }
-
-    throw new ReuniteApiError(
-      `${responseBody.title || response.statusText || 'Unknown error'}.`,
-      response.status
-    );
-  }
+  constructor(protected version: string, protected command: string) {}
 
-  protected request(url: string, options: FetchWithTimeoutOptions) {
+  public async request(url: string, options: FetchWithTimeoutOptions) {
     const headers = {
       ...options.headers,
       'user-agent': `redocly-cli/${this.version.trim()} ${this.command}`,
     };
 
-    return fetchWithTimeout(url, {
+    const response = await fetchWithTimeout(url, {
       ...options,
       headers,
     });
+
+    this.collectSunsetWarning(response);
+
+    return response;
+  }
+
+  private collectSunsetWarning(response: Response) {
+    const sunsetTime = this.getSunsetDate(response);
+
+    if (sunsetTime) {
+      const sunsetDate = new Date(sunsetTime);
+      let sunsetWarning: SunsetWarning;
+
+      if (sunsetTime > Date.now()) {
+        sunsetWarning = {
+          sunsetDate,
+          isSunsetExpired: false,
+          warning: `Endpoint ${
```

Comment:
Do we actually use the warnings somewhere? Looks like it duplicates this one: https://github.com/Redocly/redocly-cli/pull/1677/files#diff-6c6d034426b6f96932bea21f0920f55c750b51dab2f7544c4154880f3cae2182R361